pkg/grpc/interceptor/auth/githubvalidator.go

```diff
@@ -0,0 +1,70 @@
+package auth_interceptor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+const (
+	Audience               = "hookd"
+	GithubOIDCDiscoveryURL = "https://token.actions.githubusercontent.com/.well-known/jwks"
+)
+
+type GithubValidator struct {
+	jwkCache *jwk.Cache
+}
+
+func NewGithubValidator() (*GithubValidator, error) {
+	g := &GithubValidator{}
+	if err := g.setupJwkAutoRefresh(); err != nil {
+		return nil, fmt.Errorf("setup jwk auto refresh: %w", err)
+	}
+	return g, nil
+}
+
+func (g *GithubValidator) Validate(token string) (jwt.Token, error) {
+	pubKeys, err := g.jwkCache.Get(context.Background(), GithubOIDCDiscoveryURL)
```

Comment:
Should use a passed context, so that these requests time out if the caller disconnects.
This can be abused to starve hookd of resources.

As hookd is public facing (and anyone can call it and reach this code path) we need to be extra careful with code like this.